src/ert/analysis/_es_update.py

```diff
@@ -537,15 +516,34 @@ def analysis_ES(
                 f"No active observations for update step: {update_step.name}."
             )
 
-        smoother = ies.ES()
+        inversion_types = {0: "exact", 1: "subspace", 2: "subspace", 3: "subspace"}
+        inversion_type = inversion_types.get(module.ies_inversion, "Invalid input")
```

Comment:
I would prefer to fail with
```python
try:
    inversion_type = inversion_types[module.ies_inversion]
except KeyError:
    raise ...
```
Instead of passing the string `Invalid input` down into `ESMDA` and then failing there. That way we raise when the error actually occurs instead of propagating a string down which we know will fail.

Reply:
Thanks, this is much better!